src/interfaces/resource-interfaces/resource-interface.ts

```diff
@@ -1,6 +1,25 @@
-import { CertifyStatementIF } from '@/interfaces'
-
 // Interface to define the resource model example
+import { IncorporationAgreementTypeIF } from '@/interfaces'
+
 export interface ResourceIF {
-  certifyStatementResource: CertifyStatementIF | null
+  entityType: string,
+  displayName: string
+  title: string
+  description: string
+  statement: string
+  nameRequestType: string
+  directors: {
+    countMinimum: number
+  }
+  shareClasses: {
+    countMinimum: number
+  }
+  incorporationAgreement: Array<IncorporationAgreementTypeIF>,
+  reviewAndConfirm: {
+    completingPartyStatement: {
+      certifyStatementHeader: string
+      certifyStatements: Array<string>
+      certifyClause: string
+    }
+  }
```

Comment:
OK, yes I like that all differences are identified here and none are optional. This will force us to specify all these properties in the resource files, which will avoid inherited/default resources, which I think is a good thing.

For properties that are not used in a specific resource file, I think we should assign it a null/0 value and add a comment that it's not used for the subject entity type. I'll add an example below.

Reply:
Will do for sure!